lib/ansible/modules/network/onyx/onyx_facts.py

```diff
@@ -196,25 +191,38 @@ def populate(self):
 
 class Interfaces(FactsBase):
 
-    COMMANDS = ['show interfaces ethernet']
+    COMMANDS = ['show version', 'show interfaces ethernet']
```

Comment:
I can see onyx has a [cliconf plugin](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/cliconf/onyx.py#L65) that retrieves the target version at the time of initial connection.
You can refer to `get_capabilities()` [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/ios/ios.py#L82)